eth/downloader/downloader.go

```diff
@@ -1017,7 +1031,7 @@ func (d *Downloader) findAncestor(p *peer) (uint64, error) {
 					continue
 				}
 				// Otherwise check if we already know the header or not
-				if (d.mode != LightSync && d.hasBlockAndState(headers[i].Hash())) || (d.mode == LightSync && d.hasHeader(headers[i].Hash())) {
+				if (d.mode == FullSync && d.hasBlockAndState(headers[i].Hash())) || (d.mode != FullSync && d.hasHeader(headers[i].Hash())) {
```

Comment:
Why did you switch this around? Is there any difference in this change I'm not seeing?